QEfficient/utils/_utils.py

```diff
@@ -119,16 +117,16 @@ def onnx_exists(model_name: str) -> Tuple[bool, str, str]:
     return onnx_exists_bool, onnx_dir_path, onnx_model_path
 
 
-def load_hf_tokenizer(model_name: str, cache_dir: Optional[str] = None, hf_token: Optional[str] = None, padding_side:str = "left", **kwargs) -> Union[PreTrainedTokenizerFast, PreTrainedTokenizer]:
+def load_hf_tokenizer(model_name: str, cache_dir: Optional[str] = None, hf_token: Optional[str] = None, model_path: Optional[str] = None, padding_side:str = "left", **kwargs) -> Union[PreTrainedTokenizerFast, PreTrainedTokenizer]:
     logger.info(f"Loading Tokenizer for {model_name}")
     if hf_token is not None:
         login(hf_token)
 
     # Download tokenizer along with model if it doesn't exist
-    model_hf_path = hf_download(repo_id=model_name, cache_dir=cache_dir, allow_patterns=["*.json", "*.py", "*token*"])
+    model_hf_path = model_path if model_path else hf_download(repo_id=model_name, cache_dir=cache_dir, ignore_patterns=["*.txt", "*.onnx", "*.ot", "*.md", "*.tflite", "*.pdf", "*.msgpack", "*.h5"])
```

Comment:
But we do need the *.bin or *.safetensors file right? 

We have allow_patterns at one place and ignore_patterns at other. Is that by choince?

When I was trying to run Qwen/1.5 I got into error because the .safetensors file was not downloaded. The code ahead just checks if the directory with the model_name exist in the cache_dir or not. If it does then the hf_download function is not called and an error is raised for missing model files

Reply:
this is specific to `load_tokenizer` api. user need not necessarily download the checkpoints, when he wants to load tokenizer only, they call `hf_download()`. Check `infer `and `export/execute` api on how this is handled.

Reply:
I get your idea now. If someone already has the onnx or qpcs and just want the tokenizer this makes things much faster. 

How do we plan to handle the cases where there are both .safetensors and .bin files?

Reply:
I think it's fine for now to download both `.safetensors` and `.bin` files,  if we don't have a solution for this now.

Reply:
Agreed
Will remove this change in next revision